apps/angular/55-back-button-navigation/src/app/dialog.service.ts

```diff
@@ -0,0 +1,109 @@
+import { ComponentType } from '@angular/cdk/portal';
+import { Injectable, Injector, inject } from '@angular/core';
+import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
+import { take, tap } from 'rxjs';
+import {
+  DialogStrategy,
+  DialogStrategyMap,
+  DialogStrategyType,
+} from './dialog-strategy';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class DialogService {
```

Comment:
Nice service, a bit out of the challenge, but great generic service for people to add to their projects 👍